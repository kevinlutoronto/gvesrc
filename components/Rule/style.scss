@import "../../styles/base";

.wrapper {
  width: 100%;
  box-sizing: border-box;
  border: 0;
  border-top: 1px solid $cBorder;
  margin: 0;
  margin-bottom: 20px;
  background: none;
  height: 1px;
}

.withTopMargin {
  margin-top: 20px;
}

.dotted {
  border-top: 1px dotted $cBorder;
}

.dashed {
  border-top: 1px dashed $cBorder;
}
