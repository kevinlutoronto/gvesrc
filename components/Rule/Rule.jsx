// @flow
import cx from 'classnames';
import React from 'react';
import style from './style.scss';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';

const type = { SOLID: 'solid', DOTTED: 'dotted', DASHED: 'dashed' };

type Props = {|
  type?: string,
  withTopMargin?: boolean,
  data?: Data,
|};

class Rule extends React.Component<Props> {
  static defaultProps = {
    type: type.SOLID,
  };

  render() {
    const ruleStyle = cx(style.wrapper, {
      [style.dotted]: this.props.type === type.DOTTED,
      [style.dashed]: this.props.type === type.DASHED,
      [style.withTopMargin]: this.props.withTopMargin,
    });

    return <hr {...getDataAttrs(this.props)} className={ruleStyle} />;
  }
}

Rule.displayName = 'Plasma@Rule';

export default Rule;
