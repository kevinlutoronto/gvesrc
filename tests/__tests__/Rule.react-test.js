/* eslint-disable */

import React from 'react';
import Rule from '../../components/Rule/Rule';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Rule Component', () => {
  test('render', () => {
    const component = renderer.create(<Rule />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
