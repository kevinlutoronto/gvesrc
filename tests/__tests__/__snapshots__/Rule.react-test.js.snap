// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rule Component render 1`] = `
<hr
  className="wrapper"
  data-type="solid"
/>
`;
